varya/assets/sass/blocks/layout-grid/_editor.scss

```diff
@@ -1,30 +1,106 @@
+		// This padding adds end gutters.
+		padding-left: var(--layout-grid--gutter-#{$size});
+		padding-right: var(--layout-grid--gutter-#{$size});
+
+		// This padding removes end gutters.
+		&.wp-block-jetpack-layout-gutter__nowrap {
+			padding-left: 0;
+			padding-right: 0;
+		}
+	}
+}
+
+/* Padding Options */
 .wp-block-jetpack-layout-grid {
-    /* wpcom-overlay-grid is the classname targeting the grid overlay visual aid displayed in the editor */
-    .wpcom-overlay-grid {
-        grid-gap: var(--layout-grid--gutter-large) ;
-        padding-left: var(--layout-grid--gutter-large);
-        padding-right: var(--layout-grid--gutter-large);
-    }
-
-    @each $size in $layout-gutter-sizes {
-        &.wp-block-jetpack-layout-gutter__#{$size} {
-            > .block-editor-inner-blocks > .block-editor-block-list__layout,
-            .wpcom-overlay-grid {
-                grid-gap: var(--layout-grid--gutter-#{$size});
-            }
-            .wpcom-overlay-grid {
-                padding-left: var(--layout-grid--gutter-#{$size});
-                padding-right: var(--layout-grid--gutter-#{$size});
-            }
-        }
-    }
-
-    &.wp-block-jetpack-layout-gutter__nowrap {
-        .wpcom-overlay-grid {
-            padding-left: 0;
-            padding-right: 0;
-        }
-    }
+	padding-left: var(--layout-grid--gutter-large);
+	padding-right: var(--layout-grid--gutter-large);
+
+	/* Individual Column Options */
+	.wp-block-jetpack-layout-grid-column {
+		// Background Offset
+		// When a column has a background color, we add negative margins to enable
+		// some of the gutter to work as default padding.
+		&.has-background,
+		&[style^="background-color"] {
+			margin-left: calc(var(--layout-grid--background-offset) * -1);
+			margin-right: calc(var(--layout-grid--background-offset) * -1);
+			padding-left: var(--layout-grid--background-offset);
+			padding-right: var(--layout-grid--background-offset);
+		}
+	}
+
+	&.wp-block-jetpack-layout-gutter__nowrap {
+		padding-left: 0;
+		padding-right: 0;
+	}
+}
+
+/* Additional, user-set paddings. */
+// Apply both gutter padding and user-set padding, when a background is also set.
+@each $size in $layout-gutter-sizes {
+	.wp-block-jetpack-layout-grid .wp-block-jetpack-layout-grid-column.wp-block-jetpack-layout-grid__padding-#{$size} {
+		padding: var(--layout-grid--gutter-#{$size});
+
+		&.has-background,
+		&[style^="background-color"] {
+			padding-top: var(--layout-grid--gutter-#{$size});
+			padding-right: calc(var(--layout-grid--gutter-#{$size}) + var(--layout-grid--background-offset));
+			padding-bottom: var(--layout-grid--gutter-#{$size});
+			padding-left: calc(var(--layout-grid--gutter-#{$size}) + var(--layout-grid--background-offset));
+		}
+	}
+}
+
+/* Overlay grid */
+.wp-block-jetpack-layout-grid {
+	/* wpcom-overlay-grid is the classname targeting the grid overlay visual aid displayed in the editor */
+	.wpcom-overlay-grid {
+		grid-gap: var(--layout-grid--gutter-large);
```

Comment:
Just noting that the way the plugin works now is that it defaults to the large gutter size. If this ever changes, our styles would break.

Reply:
Yeah this (and many other things) will likely change quite a bit as it grows and matures and Varya will have to grow with it. The good part about it is that we’d only have to fix it here once and every child theme we build on the system will inherit the fix mostly automatically :-) 

I also thing the Sass can be audited/cleaned up even further so its easier to adapt to changes too.